userspace/libsinsp/event.h

```diff
@@ -236,9 +237,23 @@ class SINSP_PUBLIC sinsp_evt : public gen_event
 	/*!
 	  \brief Return the event category.
 	*/
+	/// TODO: in the next future we need to rename this into `get_syscall_category_from_event`
 	inline ppm_event_category get_category() const
 	{
-		return m_info->category;
+		/* Every event category is composed of 2 parts:
+		 * 1. The highest bits represent the event category:
+		 *   - `EC_SYSCALL`
+		 *   - `EC_TRACEPOINT
+		 *   - `EC_PLUGIN`
+		 *   - `EC_METAEVENT`
+		 * 
+		 * 2. The lowest bits represent the syscall category
+		 * to which the specific event belongs.
+		 * 
+		 * This function removes the highest bits, so we consider only the syscall category.
+		 */
+		const int bitmask = EC_SYSCALL - 1;
+		return static_cast<ppm_event_category>(m_info->category & bitmask);
```

Comment:
should plugin events return anything special or are we good?

Reply:
a plugin event should return `EC_OTHER` as before